src/NServiceBus.Core/Transports/Msmq/MsmqMessageDispatcher.cs

```diff
@@ -62,7 +62,9 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
                 {
                     using (var toSend = MsmqUtilities.Convert(message, transportOperation.DeliveryConstraints))
                     {
-                        toSend.UseDeadLetterQueue = settings.UseDeadLetterQueue;
+                        var ttbrRequested = toSend.TimeToBeReceived < MessageQueue.InfiniteTimeout;
+
+                        toSend.UseDeadLetterQueue = settings.UseDeadLetterQueue && (!ttbrRequested || settings.UseDeadLetterQueueForMessagesWithTimeToReachQueue);
```

Comment:
shouldn't this enable DLQ even when `UseDeadLetterQueue` has been disabled for other messages?
Also, from a readability perspective, I'd prefer a || instead of && anyway if possible.